### PR TITLE
feat/AB#78949_possibility_to_inject_aggregation_in_html_widgets

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Feat: Added possibility to inject aggregation result in html widgets.

## Useful links

[ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/78949)

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

It has been tested verifying if it's possible to select an aggregation and use it in the autocompleter in the editor in html widgets.

## Screenshots

Peek 14-11-2023 09-18

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MwanPygmay — COMMENTED** [review]
- Tested, indeed works, but options available for now are very restricted (the aggregation just appears in the propositions, but you cannot get data from it).
- Added mapping for the new aggregation

File: libs/shared/src/lib/components/widgets/common/aggregation-settings/aggregation-settings.component.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      query: GET_RESOURCES,
+      variables: {
+        first: ITEMS_PER_PAGE,
+        sortField: 'name',
+      },
+    });
+
+    this.formGroup
+      .get('aggregation.resource')
+      ?.valueChanges.subscribe((val: any) => {
```
Comment (unai-reliefapp) [inline libs/shared/src/lib/components/widgets/common/aggregation-settings/aggregation-settings.component.ts]:
```suggestion
      ?.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((val: any) => {
```

File: libs/shared/src/lib/components/widgets/common/aggregation-settings/aggregation-settings.component.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      },
+    });
+    dialogRef.closed.pipe(takeUntil(this.destroy$)).subscribe((value) => {
+      if (value) {
+        if (typeof value === 'string') {
+          this.formGroup.get('aggregation.id')?.setValue(value);
+        } else {
+          this.formGroup.get('aggregation.id')?.setValue((value as any).id);
+          this.formGroup.get('aggregation.name')?.setValue((value as any).name);
+          this.selectedAggregation = value;
```
Comment (unai-reliefapp) [inline libs/shared/src/lib/components/widgets/common/aggregation-settings/aggregation-settings.component.ts:197]: If value type set from the modal is string(aggregation id) we still have an aggregation selected in the formgroup, but the `selectedAggregation` property could be undefined, has this been thoroughly checked?(on canceling, saving and opening again modal, on switching tabs, on close and openning settings etc etc)

e.g. `editAggregation` method won't work as it's checking `if (value && this.selectedAggregation)` and if the first added aggregation only sets the id like it's done here, `selectedAggregation` is not updated and even if an aggregation id is set, cannot be edited

KR, Unai

File: libs/shared/src/lib/components/widgets/editor-settings/editor-settings.component.ts [hunk 3]
```diff
@@ -122,6 +142,13 @@ export class EditorSettingsComponent implements OnInit, AfterViewInit {
           }
         });
     }
+
+    // update fields when change aggregation
+    this.widgetFormGroup
+      .get('aggregation')
+      ?.valueChanges.subscribe((val: any) => {
```
Comment (unai-reliefapp) [inline libs/shared/src/lib/components/widgets/editor-settings/editor-settings.component.ts]:
```suggestion
      ?.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((val: any) => {
```

**Review by unai-reliefapp — CHANGES_REQUESTED** [review]

**Comment by AntoineRelief** [comment]
@RenzoPrats @unai-reliefapp @MwanPygmay 
I will need to close this one, as it won't work
my bad for this one, I didn't provided enough information in the ticket

I was expecting to have the possibility to put many aggregations, like that:

{{aggregation(id)[0].count}} for example, which should be transformed when loading the widget
this last part is missing

Also, one aggregation per widget is not enough, and it should be possible to use more than one
In addition, I'll remove the possibility to do it in summary cards, as I think it will generate conflicts

**Pull request closed without merging**